src/ucs/datastruct/array.h

```diff
@@ -143,7 +136,8 @@ BEGIN_C_DECLS
     { \
         (_array)->buffer   = (_buffer); \
         (_array)->length   = 0; \
-        (_array)->capacity = ucs_array_init_fixed_capacity(_capacity); \
```

Comment:
minor, not strictly related to patch, could add `ucs_assertv((_array)->capacity == _capacity)` after assignment, also after call to `ucs_array_grow()` in same file.

Reply:
Right, I see what you mean. Will add those checks where makes sense